src/runtime/contrib/thrust/thrust.cu

```diff
@@ -65,6 +65,105 @@ void thrust_sort(DLTensor* input,
   }
 }
 
+// Performs sorting along axis -1 and returns both sorted values and indices.
+template<typename DataType, typename IndicesType>
+void thrust_sort_nms(DLTensor* input,
+                 DLTensor* valid_count,
+                 DLTensor* out_values,
+                 DLTensor* out_indices,
+                 bool is_ascend) {
+  thrust::device_ptr<IndicesType> valid_count_ptr(static_cast<IndicesType *>(valid_count->data));
+  thrust::device_ptr<DataType> data_ptr(static_cast<DataType *>(input->data));
+  thrust::device_ptr<DataType> values_ptr(static_cast<DataType *>(out_values->data));
+  thrust::device_ptr<IndicesType> indices_ptr(static_cast<IndicesType *>(out_indices->data));
+
+  int n_values = input->shape[input->ndim - 1];
+  int n_iter = 1;
+  for (int i = 0; i < input->ndim - 1; ++i) {
+    n_iter *= input->shape[i];
+  }
+
+  thrust::copy(data_ptr, data_ptr + n_iter * n_values, values_ptr);
+
+  for (int i = 0 ; i < n_iter; ++i) {
+    int current_values = valid_count_ptr[i];
```

Comment:
if we replace `current_values` with `n_values`, the sorting code above and below looks identical with `thrust_sort`

Reply:
`thrust_sort_nms` needs one more input tensor than `thrust_sort`, I'm not sure how to pass the `valid_count` tensor without creating a new function.

Reply:
I imagine we can create another function `thrust_sort_common`, and put all common code in it. Then have existing `thrust_sort` and `thrust_sort_nms` dispatch into it appropriately.

Reply:
`thrust_sort_common` should take a function (lambda) as an argument, which takes current iter (`i` in the code) and returns:

*  For existing `thrust_sort`, always return `n_values`
*  For `thrust_sort_nms`, returns `valid_count_ptr[i]` (`current_values`)

Does this make sense?